test/framework/src/main/java/org/elasticsearch/test/AbstractXContentTestCase.java

```diff
@@ -85,7 +86,7 @@ public final void testFromXContent() throws IOException {
      * called multiple times during test execution and should return a different
      * random instance each time it is called.
      */
-    protected abstract T createTestInstance();
+    protected abstract T createTestInstance() throws IOException;
```

Comment:
I think I would prefer to catch the exception in createTestInstance instead. We should not need this in most of the cases, and it would be nice to keep this aligned with `AbstractWireTestCase#createTestInstance`